src/ephemeris.c

```diff
@@ -253,9 +245,22 @@ u8 ephemeris_valid(const ephemeris_t *eph, const gps_time_t *t)
   /* Seconds from the time from ephemeris reference epoch (toe) */
   double dt = gpsdifftime(t, &eph->toe);
 
+  /*
+   *Ephemeris did not get time-stammped when it was received.
+   */
+  if (eph->toe.wn == 0) {
+    return 0;
+  }
+
   /* TODO: this doesn't exclude ephemerides older than a week so could be made
    * better. */
-  return (eph->valid && fabs(dt) < ((u32)eph->fit_interval)*60*60);
+  /* If dt is greater than fit_interval hours our ephemeris isn't valid. */
```

Comment:
I don't think we should warn in this function if the ephemeris is invalid. It's ok to pass outdated ephemeris to this function.